src/lib/common/selector.py

```diff
@@ -34,15 +36,22 @@ def load(self):
         """ the select DF is loaded from the appropriate file """
         return pd.read_csv(self.CSV, encoding="utf-8")
 
-    def index_complete(self, df, logs):
-        """ the data (list of lists) is saved in the appropriate file """
-        # TODO: do checks to ensure a valid structure, information rich enough.
+    def index(self, config):
+        df = self.index_files(config)
         if df is not None:
             df.to_csv(self.SELECT_MAP)
-        save_logs(logs, self.SELECT_LOGS)
+        save_logs(self.__selectLogs, self.SELECT_LOGS)
+
+    def select_logger(self, msg):
+        self.__selectLogs.append(msg)
+        print(msg)
+
+    def retrieve_logger(self, msg):
+        self.__retrieveLogs.append(msg)
+        print(msg)
```

Comment:
Minor thing: you could make `__logger` a private function and then have each of these proxy it to make this slightly DRYer-- and maybe also have logs as a dict or smthn:
```python
self.logs = {
    "retrieve": [],
    "index": [],
}

# ... 

def __logger(self, key, msg):
    self.logs[key].append(msg)
    print(msg)

def retreive_logger(self, msg):
    self.__logger('retrieve', msg)

def index_logger(self, msg):
   self.__logger('index', msg)
```
```

Reply:
We could even maybe do some magic so that the selector implementation only has to call `self.logger`, and the ABC works out whether to append to retrieve or index logs based on the invocation's origin.

Reply:
Implemented as per comment - selector implementations call `self.logger` and the ABC works out the rest.